src/doom/v_snow.c

```diff
@@ -0,0 +1,121 @@
+//
+// Copyright(C) 1993-1996 Id Software, Inc.
+// Copyright(C) 2005-2014 Simon Howard
+// Copyright(C) 2015-2018 Fabian Greffrath
+//
+// This program is free software; you can redistribute it and/or
+// modify it under the terms of the GNU General Public License
+// as published by the Free Software Foundation; either version 2
+// of the License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// DESCRIPTION:
+//	[crispy] Snow drawing
+//
+
+#include <stdlib.h>
+#include <stdbool.h>
+
+#include "v_snow.h"
+#include "i_video.h"
+#include "i_video.h"
+#include "z_zone.h"
+#include "r_main.h"
+
+typedef struct snowflake_t {
+    int x, y;
+} snowflake_t;
+
+static snowflake_t     *snowflakes = NULL;
+static size_t          snowflakes_num;
+static int             snowflakes_color = -1;
+static int             last_screen_size;
+static int             wind = 1;
+
+static bool CoordsNeedsUpdate()
+{
+    static int 	last_tic = -1;
+    int		current_tic;
+
+    current_tic = I_GetTime();
+
+    if (last_tic == -1)
+	last_tic = current_tic;
+    if (current_tic == last_tic)
+	return false;
+
+    last_tic = current_tic;
+    return true;
+}
+
+static void InitSnowCoords()
+{
+    for (size_t i = 0; i < snowflakes_num; i++)
+    {
+	snowflakes[i].y = 0 - rand() % (SCREENHEIGHT*2);
+	snowflakes[i].x = rand() % (SCREENWIDTH);
+    }
+}
+
+static void UpdateSnowCoords()
+{
+    if (rand()%150 == 1)
+	wind = 1 - rand()%3;
+
+    for (size_t i = 0; i < snowflakes_num; i++)
```

Comment:
Indentation is somehow messed up in this file. Could you please just tab-indent all the new code.